shared/actions/settings.js

```diff
@@ -77,6 +89,24 @@ function * _onUpdatePGPSettings (): SagaGenerator<any, any> {
   }
 }
 
+function * _onSubmitNewEmail (): SagaGenerator<any, any> {
+  try {
+    const selector = (state: TypedState) => state.settings.email
```

Comment:
have the selector return newEmail instead. Since it's typed, you are sure you'll get newEmail